src/block-management/block-notifier.js

```diff
@@ -3,18 +3,21 @@
 var Notifier = require("./notifier.js");
 var PollingBlockNotifier = require("./polling-block-notifier.js");
 var SubscribingBlockNotifier = require("./subscribing-block-notifier.js");
+var isMetaMask = require("../utils/is-meta-mask");
 
 function BlockNotifier(transport, pollingIntervalMilliseconds) {
   var blockNotifier;
   Notifier.call(this);
-
-  blockNotifier = new SubscribingBlockNotifier(transport, function () {
-    blockNotifier.destroy();
+  if (isMetaMask()) { // MetaMask doesn't throw an error on eth_subscribe, but doesn't actually send block notifications, so we need to poll instead...
     blockNotifier = new PollingBlockNotifier(transport, pollingIntervalMilliseconds);
-    blockNotifier.subscribe(this.notifySubscribers);
-  }.bind(this));
+  } else {
+    blockNotifier = new SubscribingBlockNotifier(transport, function () {
+      blockNotifier.destroy();
+      blockNotifier = new PollingBlockNotifier(transport, pollingIntervalMilliseconds);
+      blockNotifier.subscribe(this.notifySubscribers);
+    }.bind(this));
+  }
   blockNotifier.subscribe(this.notifySubscribers);
```

Comment:
Will this double-subscribe if it goes via the non-metamask but failed subscribe path?

Reply:
Yep, but the first block notifier is destroyed, so it should be fine.  (It worked this way before, too.)

Reply:
oh i see, yes, happens later  in the callback 👍